tests/Fixer/ControlStructure/YodaStyleFixerTest.php

```diff
@@ -0,0 +1,266 @@
+            ),
+            array(
+                '<?php return null == $a[2];',
+                '<?php return $a[2] == null;',
+            ),
+            array(
+                '<?php return "" === $this->myArray[$index];',
+                '<?php return $this->myArray[$index] === "";',
+            ),
+            array(
+                '<?php return "" === $this->myArray[$index]->a;',
+                '<?php return $this->myArray[$index]->a === "";',
+            ),
+            array(
+                '<?php return "" === $this->myObject->{$index};',
+                '<?php return $this->myObject->{$index} === "";',
+            ),
+            array(
+                '<?php return "" === $this->myObject->{$index}->a;',
+                '<?php return $this->myObject->{$index}->a === "";',
+            ),
+            array(
+                '<?php return self::MY_CONST === self::$myVariable;',
+                '<?php return self::$myVariable === self::MY_CONST;',
+            ),
+            array(
+                '<?php return \A\B\C::MY_CONST === \A\B\C::$myVariable;',
+                '<?php return \A\B\C::$myVariable === \A\B\C::MY_CONST;',
+            ),
+            array(
+                '<?php return A\B::MY_CONST === B\C::$myVariable;',
+                '<?php return B\C::$myVariable === A\B::MY_CONST;',
+            ),
+            // Nested case.
+            array(
+                '<?php return null === $a[0 === $b ? $c : $d];',
+                '<?php return $a[$b === 0 ? $c : $d] === null;',
+            ),
+            array(
+                '<?php return null === $this->{null === $a ? "a" : "b"};',
+                '<?php return $this->{$a === null ? "a" : "b"} === null;',
+            ),
+            // complex code sample
+            array(
+                '<?php
+if ($a == $b) {
+    return null === $b ? (null === $a ? 0 : 0 === $a->b) : 0 === $b->a;
+} else {
+    if ($c === (null === $b)) {
+        return false === $d;
+    }
+}',
+                '<?php
+if ($a == $b) {
+    return $b === null ? ($a === null ? 0 : $a->b === 0) : $b->a === 0;
+} else {
+    if ($c === ($b === null)) {
+        return $d === false;
+    }
+}',
+            ),
+        );
+    }
+
+    /**
+     * @param string $expected
+     * @param string $input
+     *
+     * @dataProvider provideLessGreaterCases
+     * @requires PHP 5.6
+     */
+    public function testFixLessGreater($expected, $input)
+    {
+        $config = array(
+            'yoda' => true,
+            'operators' => array('less_and_greater'),
+        );
+
+        $this->fixer->configure($config);
+        $this->doTest($expected, $input);
+
+        $config['yoda'] = false;
+        $this->fixer->configure($config);
+        $this->doTest($input, $expected);
+    }
+
+    public function provideLessGreaterCases()
+    {
+        return array(
+            array(
+                '<?php $a = 3 <= $b;',
```

Comment:
> you can even config the fixer to change only some statements so exclude <, >, <=, => if you want

That's a great feature, I missed that. I personally think that Yoda is a bad practice, but if I were forced to use it in a project where I contribute (like the sonata libs), I would try to have it configured with `==` only (would be great to have this as the default). So nevermind this comment.

> CS is not always/only about logic, sometimes it really is just style

Yeah but in this particular case, no one uses that because it looks good IMO, it's just to spare accidental mistakes (so benefits the writer, but not the reader), and is pretty useless if you have unit tests.